app/lib/event/utils.ts

```diff
@@ -0,0 +1,58 @@
+import { Event } from "@prisma/client";
+import { getEventById } from "~/event.server";
+
+export async function getRootEvents(
+  events: {
+    event: Pick<Event, "parentEventId" | "name" | "slug" | "published">;
+  }[]
+) {
+  let rootEvents: typeof events = [];
+  await Promise.all(
```

Comment:
may be look for database driven hierarchical solution instead of walking one by one

Reply:
Created a ticket:
see #379

Reply:
Resolved in 09f91ba